packages/dds/matrix/src/test/memory/sharedMatrix.spec.ts

```diff
@@ -3,27 +3,124 @@
  * Licensed under the MIT License.
  */
 
+import assert from "node:assert";
+
 import {
 	type IMemoryTestObject,
 	benchmarkMemory,
 	isInPerformanceTestingMode,
 } from "@fluid-tools/benchmark";
 
 import { SharedMatrix } from "../../index.js";
+import { UndoRedoStackManager } from "../undoRedoStackManager.js";
 import { createLocalMatrix } from "../utils.js";
 
+/**
+ * Creates a benchmark for undo operations on a SharedMatrix.
+ */
+function createUndoBenchmark({
+	title,
+	matrixSize,
+	initialValue,
```

Comment:
This is the initial _cell_ value, right? Probably worth making the name more specific.